### ISSUE TITLE
API to manage u32 as bits

### ISSUE THREAD

**Issue by gallottino** [RIP-Comm/clementine#33]
Discussion [here](https://discord.com/channels/919139369774891088/1016790105417072691).

**Comment by AlessandroGrassi99** [comment]
I report and translate what I wrote in the discussion:

```Rust
fn get_bit(index: u8) -> u8;
fn get_bits(positions: &[u8]) -> u32;
fn get_bits_range(range: Range<u8>) -> u32;

fn check_bit(index: u8, value: u8) -> bool;
fn check_bits(positions: &[u8], values: &[u8]) -> bool;
fn check_bits_range(range: Range<u8>, values &[u8]) -> bool;
```

Regarding the use of a `u8` type instead of a boolean on `get_bit()` (`get_bit(..) -> u8` vs `get_bit(..) -> bool`)... For me reading `if opcode.get_bit(4) { }` is much less intuitive than reading `if opcode.get_bit(4) == 1 { }`. The same issue is even more pronounced in the matches. I think it is very important to make explicit what values we are looking for.

In addition, why put `get_bits_range()` if there is `get_bits()`?

Imho it is Rust's syntax that helps in readability because if we need bits 11 to 7, the difference is `opcode.get_bits_range(11..=7)` vs `opcode.get_bits(&(11..=7).collect::<Vec<u8>>())` or `opcode.get_bits(&vec![11,10,9,8,7])`

On small ranges (`opcode.get_bits(&vec![11,10,9,8,7])`) is readable, but in instructions like Branch, where the required offset ranges from 23 to 0 is tough ahah.

**Comment by FedericoBruzzone** [comment]
I think this issue could be closed :)

**Comment by gallottino** [comment]
Closed with #39
